### PR TITLE
feat: add shutdown for standalone and metasrv

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add `shutdown` method for standalone and metasrv

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1158

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

File: src/cmd/src/cli.rs [hunk 1]
```diff
@@ -31,7 +31,6 @@ impl Instance {
     }
 
     pub async fn stop(&self) -> Result<()> {
-        // TODO: handle cli shutdown
```
Comment (killme2008) [inline src/cmd/src/cli.rs:34]: Is it done?

File: src/cmd/src/standalone.rs [hunk 2]
```diff
@@ -152,7 +155,17 @@ impl Instance {
...
+        self.frontend
+            .shutdown()
+            .await
+            .context(ShutdownFrontendSnafu)?;
+
+        self.datanode
+            .shutdown_instance()
+            .await
+            .context(ShutdownDatanodeSnafu)?;
+        info!("Datanode instance shutdowned");
```
Comment (killme2008) [inline src/cmd/src/standalone.rs:167]:
```suggestion
        info!("Datanode instance stopped.");
```

File: src/meta-srv/src/bootstrap.rs [hunk 3]
```diff
@@ -44,44 +46,66 @@ pub struct MetaSrvInstance {
...
     let listener = TcpListener::bind(bind_addr)
         .await
         .context(error::TcpBindSnafu { addr: bind_addr })?;
     let listener = TcpListenerStream::new(listener);
 
     router
-        .serve_with_incoming(listener)
+        .serve_with_incoming_shutdown(listener, async {
+            signal.recv().await;
+            info!("Receiving shutdown signal");
```
Comment (killme2008) [inline src/meta-srv/src/bootstrap.rs:107]: The log is meaningless; please delete it.

File: src/cmd/src/cli.rs (hunk 1 above)
Comment (WenyXu) [reply src/cmd/src/cli.rs:34]: > Is it done?

Yep, the repl handled it.
https://github.com/GreptimeTeam/greptimedb/blob/f712f978cfccb7ef0b95486bc49a2155d1a9d502/src/cmd/src/cli/repl.rs#L105-L118

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**